Show SP name on "No connection between institution and service" page

pablothedude opened the pull request:
When a user wants to log into a service that the IdP does not allow,
and the user is presented with an error page "No connection between
institution and service", the entityID of the SP is listed. Now also
the SP name is shown in this message because that's more meaningful.

Also see the [Pivotal issue](https://www.pivotaltracker.com/story/show/149064131)

MKodde commented on src/OpenConext/EngineBlockBundle/Controller/FeedbackController.php:
```diff
@@ -309,6 +317,11 @@ public function unknownPreselectedIdpAction(Request $request)
+
+        $serviceProvider = $this->getServiceProviderById();
+        $name = $serviceProvider->getDisplayName($request->getLocale());
+        $customFeedbackInfo['Service Provider Name'] = $name;
```
Using 'Service Provider Name' as the title here results in a missing translation error (see screenshot)

MKodde commented on src/OpenConext/EngineBlockBundle/Controller/FeedbackController.php:
```diff
@@ -340,4 +353,16 @@ private function setFeedbackInformationOnSession(array $customFeedbackInfo)
     }
+
+    /**
+     * @SuppressWarnings(PHPMD.Superglobals) This is required to mimic the existing functionality
```
As we discussed IRL, not the most elegant solution to get the current SP but I think this is the easiest way to get it at this point.

Alternatively, you could fish the required information from the `SAMLRequest` index in the SESSION superglobal.

MKodde commented on src/OpenConext/EngineBlockBundle/Controller/FeedbackController.php:
```diff
@@ -31,19 +32,26 @@ class FeedbackController
         // we have to start the old session in order to be able to retrieve the feedback info
         $server = new EngineBlock_Corto_ProxyServer($twig);
         $server->startSession();
+
```
Empty line on the last line of the constructor may be removed

#nitpick

pablothedude commented:
I've added the missing translations and fixed the empty line. I've looked into the option to get the name of the service provider form the `SAMLRequest` but I wasn;t able to do so.

Also I removed the logic from the FeedbackController and moved it to the ApplicationSingleton. As a result the name of the service provider is always set if an error report is being generated. I think that's not an issue and it's always helpful on an error report when a service provider id is set but makes sure the name is updated after a new error report was generated. I will report this behavior on a ticket so @thijskh is ware of this change.
